app/uk/gov/hmrc/tai/connectors/TaxAccountConnector.scala

```diff
@@ -80,14 +78,13 @@ class TaxAccountConnector @Inject()(httpHandler: HttpHandler, servicesConfig: Se
         Left(e.getMessage)
     }
 
-  def nonTaxCodeIncomes(nino: Nino, year: TaxYear)(implicit hc: HeaderCarrier): Future[TaiResponse] =
-    httpHandler.getFromApiV2(nonTaxCodeIncomeUrl(nino.nino, year)) map (
-      json => TaiSuccessResponseWithPayload((json \ "data").as[Incomes].nonTaxCodeIncomes)
-    ) recover {
-      case e: UnauthorizedException => TaiUnauthorisedResponse(e.getMessage)
+  def nonTaxCodeIncomes(nino: Nino, year: TaxYear)(implicit hc: HeaderCarrier): Future[NonTaxCodeIncome] =
+    httpHandler.getFromApiV2(nonTaxCodeIncomeUrl(nino.nino, year)).map { json =>
+      (json \ "data").as[Incomes].nonTaxCodeIncomes
+    } recover {
```

Comment:
There is a slight difference in behavior compared to the previous code. In this case, UnauthorizedExceptions will log a warning, whereas in the previous code this would not be done.

Reply:
the question is whether we want to warn that or not